pennylane/interfaces/batch/tensorflow.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for adding the TensorFlow interface
+to a PennyLane Device class.
+"""
+# pylint: disable=too-many-arguments,too-many-branches
+import inspect
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.eager import context
+
+import pennylane as qml
+
+
+def _compute_vjp(dy, jacs):
+    vjps = []
+
+    for d, jac in zip(dy, jacs):
+        vjp = qml.gradients.compute_vjp(d, jac)
+
+        if not context.executing_eagerly():
+            vjp = qml.math.unstack(vjp)
+
+        vjps.extend(vjp)
+
+    return vjps
+
+
+def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=2):
+    """Execute a batch of tapes with TensorFlow parameters on a device.
+
+    Args:
+        tapes (Sequence[.QuantumTape]): batch of tapes to execute
+        device (.Device): Device to use to execute the batch of tapes.
+            If the device does not provide a ``batch_execute`` method,
+            by default the tapes will be executed in serial.

```

Comment:
```suggestion
            by default the tapes will be executed in series.

```

Reply:
Wait, shouldn't this be serial (in contrast to parallel)?

Reply:
Looking into it I think is the same thing. Can disregard :)